### PR TITLE
#765: Use HeapDump to get heapdump file's content from S3

### PR DESCRIPTION
**Pull request by proshin-roman**
PR for #765. 

I've changed behaviour of TkDump class: now it uses ExtBucket functionality to retrieve heapdump.perf file from S3 storage instead of using a local file.

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/zerocracy/farm/issues/1314#issuecomment-402047966))

@g4s8 Job [#1314](https://github.com/zerocracy/farm/issues/1314) is already in scope

**Comment by g4s8** [comment]
@0crat assign @carlosmiranda

**Comment by 0crat** [comment]
> @0crat assign @carlosmiranda  ([here](https://github.com/zerocracy/farm/issues/1314#issuecomment-402049927))

@g4s8 This pull request [#1314](https://github.com/zerocracy/farm/issues/1314) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1840806120); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/zerocracy/tk/TkDump.java [hunk 1]
```diff
@@ -34,27 +32,14 @@
...
-                    String.format("File doesn't exist: %s", file),
-                    Level.SEVERE
-                )
-            );
-        }
         return new RsWithType(
             new RsWithBody(
                 new BufferedInputStream(
-                    new FileInputStream(file)
+                    new HeapDump(new ExtBucket().value(), "").load()
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/tk/TkDump.java]: @proshin-roman this specific change looks good, but let's make this class more testable. Let's create `TkDumpTest` and test if it returns the heap dump. We can add a package private constructor where we can supply a different `Bucket` instance (i.e., a fake/mock for testing).

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1314](https://github.com/zerocracy/farm/issues/1314) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1314#issuecomment-403198977))

@ypshenychka This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1314](https://github.com/zerocracy/farm/issues/1314) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231314%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**